Astar.elm

```diff
@@ -232,49 +157,44 @@ processNeighbors grid open neighbors current dest =
         processNeighbors nextGrid nextOpen rest current dest
 
 
-estimateNext : Grid -> NodeList -> NodeList -> Node -> Node -> List (Int, Int)
-estimateNext grid open closed dest current =
+estimateNext : NodeList -> NodeList -> Node -> Node -> Grid -> List (Int, Int)
+estimateNext open closed dest current grid =
   let
     nextOpen = removeNode open current
     nextClosed = current :: closed
     neighbors = getNeighbors grid current
-    freeNeighbors = filter (\ n -> not n.obstacle) neighbors
-    openNeighbors = filter (\ n -> not (contains nextClosed n)) freeNeighbors
+    openNeighbors = List.filter (\ n -> not (contains n nextClosed)) neighbors
     gridAndOpen = processNeighbors grid nextOpen openNeighbors current dest
     continueGrid = fst gridAndOpen
     continueOpen = snd gridAndOpen
   in
-    astarIter continueGrid continueOpen nextClosed dest
+    astar continueOpen nextClosed dest continueGrid
 
 
-astarIter : Grid -> NodeList -> NodeList -> Node -> List (Int, Int)
-astarIter grid open closed dest =
+astar : NodeList -> NodeList -> Node -> Grid -> List (Int, Int)
+astar open closed dest grid =
   case (smallestF open) of
     Nothing -> []
     Just c ->
-      if nodeEq c dest then
+      if nodeq c dest then
         pathToTileList grid (Just c.tile)
       else
-        estimateNext grid open closed dest c
-
-
-clipFirst : List a -> List a
-clipFirst list =
-  case list of
-    [] -> []
-    _ :: rest -> rest
+        estimateNext open closed dest c grid
 
 
-astar : (Int, Int) -> List (Int, Int) -> (Int, Int) -> (Int, Int) -> List (Int, Int)
-astar gridSize obstacles start destination =
+findPath : (Int, Int) -> (Int, Int) -> Grid -> List (Int, Int)
+findPath start destination grid =
```

Comment:
I think that the order of the arguments should be  `grid start destination` because the last argument should be the most important one, e.g. you wouldn't want to try your function for different grids, but rather for different destinations from within the same grid.